static/src/js/containers/DownloadHistoryContainer/__tests__/DownloadHistoryContainer.test.js

```diff
@@ -1,89 +1,368 @@
+            created_at: '2019-08-25T11:58:14.390Z',
+            collections: [{
+              title: 'Collection Title'
+            }]
+          }],
+          retrievalHistoryLoading: false,
+          retrievalHistoryLoaded: true,
+          onDeleteRetrieval: jest.fn()
+        })
+
+        expect(screen.getByRole('table')).toBeInTheDocument()
+        const timeElement = screen.getByText(/years ago/i).closest('time')
+        expect(timeElement).toHaveAttribute('datetime', '2019-08-25T11:58:14.390Z')
+      })
+    })
+
+    describe('when a retrieval exists with no collections', () => {
+      test('renders a table with "0 collections"', () => {
+        setup({
+          earthdataEnvironment: 'prod',
+          retrievalHistory: [{
+            id: '8069076',
+            jsondata: {},
+            created_at: '2019-08-25T11:58:14.390Z',
+            collections: []
+          }],
+          retrievalHistoryLoading: false,
+          retrievalHistoryLoaded: true,
+          onDeleteRetrieval: jest.fn()
+        })
+
+        expect(screen.getByRole('table')).toBeInTheDocument()
+        expect(screen.getByRole('link', { name: '0 collections' })).toHaveAttribute('href', '/downloads/8069076')
+      })
+    })
   })
 })
 
 describe('mapStateToProps', () => {
-  test('returns the correct state', () => {
-    const store = {
-      earthdataEnvironment: 'prod',
-      retrievalHistory: {
-        history: {},
-        isLoading: false,
-        isLoaded: false
-      }
-    }
-
-    const expectedState = {
-      earthdataEnvironment: 'prod',
-      retrievalHistory: {},
-      retrievalHistoryLoading: false,
-      retrievalHistoryLoaded: false
+  test('maps state to props correctly', () => {
+    const state = {
+      authToken: 'mockToken',
+      earthdataEnvironment: 'prod'
     }
-
-    expect(mapStateToProps(store)).toEqual(expectedState)
+    const props = mapStateToProps(state)
+    expect(props).toEqual({
+      authToken: 'mockToken',
+      earthdataEnvironment: 'prod'
+    })
   })
 })
 
-describe('DownloadHistoryContainer component', () => {
-  describe('when passed the correct props', () => {
-    test('renders a table when a retrieval exists with one collection that has no title', () => {
-      const { enzymeWrapper } = setup({
-        authToken: 'testToken',
-        earthdataEnvironment: 'prod',
-        retrievalHistory: [{
-          id: '8069076',
-          jsondata: {},
-          created_at: '2019-08-25T11:58:14.390Z',
-          collections: [{}]
-        }],
-        retrievalHistoryLoading: false,
-        retrievalHistoryLoaded: false,
-        onFetchRetrievalHistory: jest.fn(),
-        onDeleteRetrieval: jest.fn()
+describe('mapDispatchToProps', () => {
+  beforeEach(() => {
+    jest.spyOn(errorsModule, 'handleError').mockImplementation(
+      (errorConfig) => ({
+        type: 'HANDLE_ERROR',
+        payload: errorConfig
       })
+    )
+  })
+
+  test('maps dispatch to props correctly', () => {
```

Comment:
This needs to be more descriptive, mapDispatchToProps tests are testing individual functions that have specific results. See any of the other tests as an example

```
  test('onChangeMap calls actions.changeMap', () => {
    const dispatch = jest.fn()
    const spy = jest.spyOn(actions, 'changeMap')

    mapDispatchToProps(dispatch).onChangeMap({ mock: 'data' })

    expect(spy).toHaveBeenCalledTimes(1)
    expect(spy).toHaveBeenCalledWith({ mock: 'data' })
  })
```